CONTRIBUTING.md

```diff
@@ -211,3 +211,42 @@ command above.
   test outputs are in one file.
   However, it may slow down the tests for not running in parallel
   and may cause the test to timeout.
+
+## Contributing to Keras applications
+
+Contributions to the
+[pre-trained application library](https://keras.io/api/applications/)
+are welcome. Code for Keras applications is located in Keras repository in
+[keras/applications](https://github.com/keras-team/keras/blob/master/keras/applications).
+When contributing to Keras applications, please keep following checklist in
+mind.
+
+ - Keras applications must implement an established and widely used model.
+   Applications should include a link to a paper describing the architecture of
+   the model with at least 20 citations.
+ - Applications should be provided with pre-trained weights.
+   - When submitting a pull request for a Keras application, these weights can
+     be provided at any publically available URL (e.g. a personal Cloud Storage
+     bucket). The weights will be uploaded to a Keras storage bucket while
+     merging the pull request.
+   - Weights should be downloaded with the
+     [get_file()](https://keras.io/api/utils/python_utils/#getfile-function)
+     utility function. Be sure to include the `file_hash` argument, which allows
+     cache invalidation on the downloaded weights. The command line programs
+     `shasum` and `sha256sum` can compute a file hash.
+ - You should help us verify that the accuracy of the model with pre-trained
+   weighted matches the reported results of the cited paper.
+ - You should add any new applications to the unit tests defined in
+   `applications_test.py` and `applications_load_weight_test.py`.
```

Comment:
No other applications do. We should probably have an example if we are going to ask for this?

Reply:
I see. Not necessarily asking for one here, but I was curious that if there's no unit test in place I wonder how people make sure their code runs.

Reply:
I believe the load weights test will load the weights, call predict, and make sure things run.

But I'm also definitely not the expert on Keras applications, so may be missing things. This is just me typing up the notes I had from doing at application review with @fchollet.

Reply:
That's right, but the test is really more a sanity check that verifies nothing is obviously wrong -- it does not check that the validation accuracy of the model matches what's expected. It would pass just as well for a 60% accuracy model and for a 75% accuracy model. This relies on the fact that the outcome of loading weights in a model is binary: either it works as expected, or nothing makes sense. It's extremely rare for the process to "most work but not quite" (the only possibility is that the data normalization step is off by a bit -- any other bug will lead to catastrophic failure).

In practice we need a one-time check (on our side) that the validation accuracy matches what's expected, after the model is submitted. We can always revert in case of major issue.